ophyd/mixins.py

```diff
@@ -0,0 +1,125 @@
+import types
+from .signal import EpicsSignal
+from .positioner import PositionerBase
+
+
+class SignalPositionerMixin(PositionerBase):
+    '''Mixin to make a Signal a Positioner
+
+    Should be mixed in first, with the Signal second, such that:
+        set, move will be from PositionerBase
+
+    Parameters
+    ----------
+    set_func : callable
+        The set() functionality for the class. Must be specified as the mixin
+        takes over set() functionality.
+    egu : str, optional
+        Engineering units of positioner
+    hold_on_stop : bool, optional
+        When stop is called on the positioner
+    '''
+    def __init__(self, *args, set_func=None, egu='', hold_on_stop=False,
```

Comment:
If you drop the `None` it will be a required kwarg

```python

In [35]: def foo(*args, k, b=5): pass

In [36]: foo('a')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-36-7407a2d90da0> in <module>()
----> 1 foo('a')

TypeError: foo() missing 1 required keyword-only argument: 'k'
```